devlib/impl/StorageDeviceFileImpl.cpp

```diff
@@ -1,6 +1,28 @@
 #include "StorageDeviceFileImpl.h"
 
 
+namespace {
+
+    bool umountDevice(const std::shared_ptr<devlib::IStorageDeviceInfo> deviceInfo)
+    {
+        if (devlib::native::umountDisk(deviceInfo->filePath())) {
+            return true;
+        }
+
+        auto mntpts = deviceInfo->mountpoints();
+        for (auto const & mntpt : mntpts) {
+            auto mntptLock = mntpt->umount();
+            if (!mntptLock->locked()) {
+                return false;
+            }
```

Comment:
> macOS platform does not allow users without admin privileges to unmount partitions separately. So, we should try to unmount the entire disk before the partitions.

Does it mean that all users are able to unmount the entire disk and no one will ever reach `mntpt->umount()` command?

Reply:
> Does it mean that all users are able to unmount the entire disk?

In general, yes

> no one will ever reach mntpt->umount() command

I think it's possible if `unmountDisk` will fail for some reasons. But during testing, I have not seen such situations.
This is why I left log messages. This will help us in debugging the code if users encounter something unusual.